go-controller/pkg/util/kube.go

```diff
@@ -99,29 +97,11 @@ func ExtractDbRemotesFromEndpoint(ep *kapi.Endpoints) ([]string, int32, int32, e
 	return masterIPList, sbDBPort, nbDBPort, nil
 }
 
-func GetNodeIP(nodeName string) (string, error) {
-	ip := net.ParseIP(nodeName)
-	if ip == nil {
-		addrs, err := net.LookupIP(nodeName)
-		if err != nil {
-			return "", fmt.Errorf("Failed to lookup IP address for node %s: %v", nodeName, err)
+func GetNodeIP(node *kapi.Node) (string, error) {
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == kapi.NodeInternalIP {
```

Comment:
To match the behavior elsewhere in kubernetes you need to look at `NodeExternalIP` addresses too if you don't find an internal one. (Really, `k8s.io/kubernetes/pkg/util/node/node.go:GetNodeHostIP` should be moved into `k8s.io/utils` or something...)

Reply:
Do you mean
```
for {
   if addr.Type == kapi.NodeInternalIP || addr.Type == kapi.NodeExternalIP { [...] }
}
```
or:
```
for {
   if addr.Type == kapi.NodeInternalIP { [...] }
}
for {
   if addr.Type == kapi.NodeExternalIP { [...] }
}
```
?

Reply:
the latter